From d4a952a714078d3dd2274367d211ef7becc16b66 Mon Sep 17 00:00:00 2001
From: Delphine CC Chiu <Delphine_CC_Chiu@wiwynn.com>
Date: Thu, 7 Mar 2024 10:41:16 +0800
Subject: [PATCH 06/22] configurations: yosemite4: Add IANA for sentinel dome

Configuring IANA for devices indicates which OEM sets we should be
using on which hardware.

schemas: Add IANA with type of string.

Use case for PLDM OEM command:
https://gerrit.openbmc.org/c/openbmc/pldm/+/69246

Change-Id: Ifcd410f6548c5f28d399c2ef64a2aadd57554e41
Signed-off-by: Delphine CC Chiu <Delphine_CC_Chiu@wiwynn.com>
---
 configurations/yosemite4_sentineldome_t1.json         | 1 +
 configurations/yosemite4_sentineldome_t1_retimer.json | 1 +
 configurations/yosemite4_sentineldome_t2.json         | 1 +
 configurations/yosemite4_sentineldome_t2_retimer.json | 1 +
 schemas/legacy.json                                   | 6 ++++++
 5 files changed, 10 insertions(+)

diff --git a/configurations/yosemite4_sentineldome_t1.json b/configurations/yosemite4_sentineldome_t1.json
index 8d951bb..9d1e2cf 100644
--- a/configurations/yosemite4_sentineldome_t1.json
+++ b/configurations/yosemite4_sentineldome_t1.json
@@ -333,6 +333,7 @@
             "Address": "0x20",
             "Bus": "$bus % 16",
             "EndpointId": "$bus % 15 * 10",
+            "IANA": "0015A000",
             "Name": "BIC",
             "Type": "MCTPEndpoint"
         }
diff --git a/configurations/yosemite4_sentineldome_t1_retimer.json b/configurations/yosemite4_sentineldome_t1_retimer.json
index 25d0b85..6d3dd29 100644
--- a/configurations/yosemite4_sentineldome_t1_retimer.json
+++ b/configurations/yosemite4_sentineldome_t1_retimer.json
@@ -352,6 +352,7 @@
             "Address": "0x20",
             "Bus": "$bus % 16",
             "EndpointId": "$bus % 15 * 10",
+            "IANA": "0015A000",
             "Name": "BIC",
             "Type": "MCTPEndpoint"
         }
diff --git a/configurations/yosemite4_sentineldome_t2.json b/configurations/yosemite4_sentineldome_t2.json
index 987c96a..7db42c9 100644
--- a/configurations/yosemite4_sentineldome_t2.json
+++ b/configurations/yosemite4_sentineldome_t2.json
@@ -392,6 +392,7 @@
             "Address": "0x20",
             "Bus": "$bus % 16",
             "EndpointId": "$bus % 15 * 10",
+            "IANA": "0015A000",
             "Name": "BIC",
             "Type": "MCTPEndpoint"
         }
diff --git a/configurations/yosemite4_sentineldome_t2_retimer.json b/configurations/yosemite4_sentineldome_t2_retimer.json
index d1fd845..df24015 100644
--- a/configurations/yosemite4_sentineldome_t2_retimer.json
+++ b/configurations/yosemite4_sentineldome_t2_retimer.json
@@ -411,6 +411,7 @@
             "Address": "0x20",
             "Bus": "$bus % 16",
             "EndpointId": "$bus % 15 * 10",
+            "IANA": "0015A000",
             "Name": "BIC",
             "Type": "MCTPEndpoint"
         }
diff --git a/schemas/legacy.json b/schemas/legacy.json
index df89e4e..ece7201 100644
--- a/schemas/legacy.json
+++ b/schemas/legacy.json
@@ -80,6 +80,9 @@
                 "GpioPolarity": {
                     "$ref": "#/definitions/Types/GpioPolarity"
                 },
+                "IANA": {
+                    "$ref": "#/definitions/Types/IANA"
+                },
                 "Index": {
                     "$ref": "#/definitions/Types/Index"
                 },
@@ -778,6 +781,9 @@
             "GpioPolarity": {
                 "enum": ["Low"]
             },
+            "IANA": {
+                "type": "string"
+            },
             "Index": {
                 "type": "number"
             },
-- 
2.25.1

